source/common/conn_pool/conn_pool_base.cc

```diff
@@ -256,12 +256,16 @@ ConnectionPool::Cancellable* ConnPoolImplBase::newStreamImpl(AttachContext& cont
     return nullptr;
   }
 
-  if (!host_->cluster().resourceManager(priority_).pendingRequests().canCreate()) {
+  auto action = host_->canCreateRequest(priority_);
+  if (action == Upstream::HostDescriptionImpl::CreateAction::FAIL) {
     ENVOY_LOG(debug, "max pending streams overflow");
     onPoolFailure(nullptr, absl::string_view(), ConnectionPool::PoolFailureReason::Overflow,
                   context);
     host_->cluster().stats().upstream_rq_pending_overflow_.inc();
     return nullptr;
+  } else if (action == Upstream::HostDescriptionImpl::CreateAction::QUEUE) {
+    ENVOY_LOG(debug, "queueing stream");
+    return newPendingStream(context, can_send_early_data);
```

Comment:
I'm a little confused. We have the existing pending stream logic right below this. Don't we want to just do what we did before? I guess I'm wondering why we need the enum. Can't we just have a bool canCreate() function which if false flows into the existing logic and checks both the global cluster and per host limits?